src/sentry/workflow_engine/migration_helpers/rule_conditions.py

```diff
@@ -0,0 +1,301 @@
+
+    return {}, [payload]
+
+
+def create_issue_category_filter(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return {}, [
+        {
+            "id": "sentry.rules.filters.issue_category.IssueCategoryFilter",
+            "value": data_condition.comparison["value"],
+        }
+    ]
+
+
+def create_issue_occurrences_filter(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return {}, [
+        {
+            "id": "sentry.rules.filters.issue_occurrences.IssueOccurrencesFilter",
+            "value": data_condition.comparison["value"],
+        }
+    ]
+
+
+def create_latest_release_filter(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return {}, [
+        {
+            "id": "sentry.rules.filters.latest_release.LatestReleaseFilter",
+        }
+    ]
+
+
+def create_latest_adopted_release_filter(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return {}, [
+        {
+            "id": "sentry.rules.filters.latest_adopted_release_filter.LatestAdoptedReleaseFilter",
+            "oldest_or_newest": data_condition.comparison["release_age_type"],
+            "older_or_newer": data_condition.comparison["age_comparison"],
+            "environment": data_condition.comparison["environment"],
+        }
+    ]
+
+
+def create_base_event_frequency_condition(
+    id: str,
+    data_condition: DataCondition,
+    count_type: Condition,
+    percent_type: Condition,
+) -> ConditionAndFilters:
+    payload = {
+        "id": id,
+        "comparisonType": (
+            ComparisonType.COUNT if data_condition.type == count_type else ComparisonType.PERCENT
+        ),
+        "interval": data_condition.comparison["interval"],
+        "value": data_condition.comparison["value"],
+    }
+
+    if data_condition.type == percent_type:
+        payload["comparisonInterval"] = data_condition.comparison["comparison_interval"]
+
+    return payload, []
+
+
+def create_event_frequency_condition(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return create_base_event_frequency_condition(
+        id="sentry.rules.conditions.event_frequency.EventFrequencyCondition",
+        data_condition=data_condition,
+        count_type=Condition.EVENT_FREQUENCY_COUNT,
+        percent_type=Condition.EVENT_FREQUENCY_PERCENT,
+    )
+
+
+def create_percent_sessions_condition(
+    data_condition: DataCondition, is_filter: bool = False
+) -> ConditionAndFilters:
+    return create_base_event_frequency_condition(
+        id="sentry.rules.conditions.event_frequency.EventFrequencyPercentCondition",
+        data_condition=data_condition,
+        count_type=Condition.PERCENT_SESSIONS_COUNT,
+        percent_type=Condition.PERCENT_SESSIONS_PERCENT,
+    )
+
+
+def create_event_unique_user_frequency_condition(
```

Comment:
this is a single Rule condition + a bunch of filters. we store it as a single DataCondition though